Specify allowed contexts in jobs.<job_id>.services.<service_id>.image (#33747)
github/docs: content/actions/using-workflows/workflow-syntax-for-github-actions.md

The Docker image to use as the service container to run the action. The value can be the Docker Hub image name or a registry name.

> > [!NOTE]

Reviewer: Is this a new pattern within this document? Wondering whether we need to duplicate this information here, since it's already [listed here](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability)

Author: Wow, did not see this section before. I can see it was committed around a month ago so this is probably why. You are right. I'll drop this.

Author: Actually if allowed context are specified clearly already. I will close this PR.

> You can use the following [AUTOTITLE](/actions/learn-github-actions/contexts) when specifying a service image: `github`, `inputs`, `matrix`, `needs`, `strategy`, and `vars`.

If `jobs.<job_id>.services.<service_id>.image` is assigned an empty string, the service will not start. You can use this to set up conditional services, similar to the following example.

```yaml
services:
  nginx:

>     image: {% raw %}${{ matrix.nginx_service == 'true' && 'nginx' || '' }}{% endraw %}

Reviewer: Wondering whether `matrix.nginx_service == true` would be better. Booleans are supported matrix values.

The Docker image to use as the service container to run the action. The value can be the Docker Hub image name or a registry name.

> [!NOTE]

> > Usage of GitHub [AUTOTITLE](/actions/learn-github-actions/contexts) is allowed when specyfing service image.
> > Allowed contexts are: `github`, `strategy`, `matrix`, `needs`, `vars` and `inputs`.

Reviewer 2: I'd suggest rewording this slightly to use the active voice.

```suggestion
> You can use the following [AUTOTITLE](/actions/learn-github-actions/contexts) when specifying a service image: `github`, `strategy`, `matrix`, `needs`, `vars`, and `inputs`.
```
